src/platform_impl/web/web_sys/canvas.rs

```diff
@@ -72,11 +73,34 @@ impl Canvas {
             MouseState::NoPointerEvent(mouse_handler::MouseHandler::new())
         };
 
+        let common = Common {
+            raw: canvas,
+            wants_fullscreen: Rc::new(RefCell::new(false)),
+        };
+
+        let disable_web_scroll = if !attr.enable_web_scroll {
+            Some([
+                common.add_event("pointermove", move |event: Event| {
+                    event.prevent_default();
+                }),
+                common.add_event("pointermove", move |event: Event| {
+                    event.prevent_default();
+                }),
+                common.add_event("touchstart", move |event: Event| {
+                    event.prevent_default();
+                }),
+                common.add_event("touchend", move |event: Event| {
+                    event.prevent_default();
+                }),
+                common.add_event("wheel", move |event: Event| {
+                    event.prevent_default();
+                }),
```

Comment:
I think I'd rather keep the `event.prevent_default()` calls inside the original event handlers, and then check a boolean in there, instead of registering new event handlers just for that.

Reply:
I'm not super familiar with web, what is the concern here?
My guess is performance but I want to check.

Reply:
I've resolved merge conflicts, so would appreciate a response here.
But also, when I get time I'll try to just rewrite as suggested, as I'd rather not stall on something small like this.

Reply:
Sorry, I'd forgotten all about this.

I think my was mostly that I don't know enough about how `preventDefault` actually works to say with certainty whether the fact that it works here is part of the spec, or accidental.

Also, it seems like in the meantime https://github.com/rust-windowing/winit/pull/2365 was merged, doesn't that solve exactly the same problem?

Reply:
This PR introduces `prevent_default` calls for many new kinds of events, so its still needed in some form.

However I think you make a good point and we should just put these new `prevent_default`s behind `with_prevent_default` instead of introducing a new `enable_web_scroll`.

I think generally an application falls into wants every prevent_default call or wants none of them.
I believe the few outliers can then manually make their own prevent_default calls directly through web_sys if really needed.
And if winit decides it needs to provide finer grained selection of its prevent_defaults, the API can always be changed later.

Does that sound good to you?

Reply:
I went ahead and did that.
The new functionality of this PR is now behind with_prevent_default.

Reply:
> I think my was mostly that I don't know enough about how preventDefault actually works to say with certainty whether the fact that it works here is part of the spec, or accidental.

I feel like the only reasonable interpretation of [preventDefault](https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault)  is that if any of the callbacks call prevent_default then the default callback handler is not run.
Additionally I am not duplicating any callbacks, so I'm not even hitting that possible edge case any more.
So I believe my implementation is sound now and will be sound even if additional event handlers are added.

That said if you want me to write it in a different way im all ears.

Reply:
I still think I would like it all to be handled inside one callback, like `prevent_default` works currently.

Reply:
I'm not quite following, these events don't have an existing callback.

Reply:
Apologies! I got confused because we do have an existing event for `pointermove`, so I would like that one to handled like the rest; but you're right that we don't have existing `touchstart`/`touchend` events, you should add those 

```rust
// Canvas
pub fn on_touch_start(&mut self, prevent_default: bool) {
    self.on_touch_start = Some(self.common.add_event("touchstart", move |event: TouchEvent| {
        if prevent_default {
            event.prevent_default();
        }
    }));
}

// EventLoopWindowTarget::register
canvas.on_touch_start(prevent_default);
canvas.on_touch_end(prevent_default);
```

Reply:
Oh my bad, I missed pointermove already had a callback.
I've made the changes as requested and verified it still functions correctly.